apps/shinkai-desktop/src/pages/settings.tsx

```diff
@@ -96,10 +103,27 @@ const SettingsPage = () => {
   const setDefaultAgentId = useSettings(
     (settingsStore) => settingsStore.setDefaultAgentId,
   );
+
   const { nodeInfo, isSuccess: isNodeInfoSuccess } = useGetHealth({
     nodeAddress: auth?.node_address ?? '',
   });
 
+  const { mutateAsync: setMaxChatIterationsMutation } = useSetMaxChatIterations({
+    onSuccess: (_data, variables) => {
+      queryClient.invalidateQueries({ queryKey: [FunctionKeyV2.GET_PREFERENCES] });
```

Comment:
could we do all of the invalidate queries inside the shinkai node state layer?

Reply:
DONE